azure-pipelines.yml

```diff
@@ -1,342 +1,424 @@
+                  tfInitSub: ${{ variables.tfInitSub }}
+                  serviceConnection: ${{ component.serviceConnection }}
+                  action: ${{ parameters.action }}
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  setClusterNumbersEnvVar: true
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_resource_group.kubernetes_resource_group["\"${{parameters.cluster}}\""]
+                      -target module.kubernetes["\"${{parameters.cluster}}\""]
 
-  - stage: PipelineTests
-    displayName: "PipelineTests"
-    dependsOn: BootStrapClusters
-    jobs:
-      - job: PipelineTests
-        pool:
-          vmImage: ${{ variables.agentPool }}
-        timeoutInMinutes: ${{ variables.timeoutInMinutes }}
+          - job: "DeployInfrastructuremis"
+            dependsOn: DeployInfrastructure
+            steps:
+              - template: pipeline-steps/deploy-service.yaml
+                parameters:
+                  environment: ${{ component.env }}
+                  location: ${{ parameters.location }}
+                  stack: "aks-mis"
+                  project: $(project)
+                  tfInitSub: ${{ variables.tfInitSub }}
+                  serviceConnection: ${{ component.serviceConnection }}
+                  action: ${{ parameters.action }}
+                  setClusterNumbersEnvVar: false
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_role_assignment.uami_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_read_rg["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_dns_zone_contributor["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview2aat_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target data.azurerm_kubernetes_cluster.kubernetes["\"${{parameters.cluster}}\""]
+
+
+      - stage: "BootStrapClusters_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: BootStrap Clusters"
+        dependsOn:
+          - "Aks_${{ component.env }}"
+        variables:
+          existing_clusters_from_aks: $[ stageDependencies.Aks_${{ component.env }}.SetClusterVariables.outputs['SetExistingClustersVariable.existing_clusters'] ]
+          deploy_cluster_from_aks: $[ stageDependencies.Aks_${{ component.env }}.SetDeployClusterVariable.outputs['SetDeployClusterVariable.deploy_cluster'] ]
```

Comment:
Again a bit confused by these variables being set, are you using them somewhere?